core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -617,22 +621,38 @@ <R> R rpc(RpcCallable<FileSystemMasterClient, R> fn)
       // Explicitly connect to trigger loading configuration from meta master.
       client.get().connect();
       return fn.call(client.get());
-    } catch (NotFoundException e) {
+    } catch (AlluxioStatusException e) {
+      wrapAndThrowAlluxioStatusException(e);
+      throw UNREACHABLE_CODE_ERROR;
+    }
+  }
+
+  protected void wrapAndThrowAlluxioStatusException(AlluxioStatusException e)
+      throws AlluxioException, IOException {
+    if (e instanceof NotFoundException) {
       throw new FileDoesNotExistException(e.getMessage());
-    } catch (AlreadyExistsException e) {
+    }
+    if (e instanceof AlreadyExistsException) {
       throw new FileAlreadyExistsException(e.getMessage());
-    } catch (InvalidArgumentException e) {
+    }
+    if (e instanceof InvalidArgumentException) {
       throw new InvalidPathException(e.getMessage());
-    } catch (FailedPreconditionException e) {
+    }
+    if (e instanceof FailedPreconditionException) {
       // A little sketchy, but this should be the only case that throws FailedPrecondition.
       throw new DirectoryNotEmptyException(e.getMessage());
-    } catch (UnavailableException e) {
-      throw e;
-    } catch (UnauthenticatedException e) {
+    }
+    if (e instanceof UnavailableException || e instanceof UnauthenticatedException) {
       throw e;
-    } catch (AlluxioStatusException e) {
-      throw e.toAlluxioException();
     }
+    throw e.toAlluxioException();
+  }
+
+  /**
```

Comment:
nit: If there's anything you prefer sharing context for (or something warnings while sharing a context), let's mention here a bit

Reply:
n/a don't think there's anything worth mentioning